testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java

```diff
@@ -119,10 +121,64 @@ public String getLogicalName()
 
     public DockerContainer withEnvironmentListener(Optional<EnvironmentListener> listener)
     {
-        this.listener = requireNonNull(listener, "listener is null");
+        listener.ifPresent(listeners::add);
         return this;
     }
 
+    public DockerContainer withContainerListener(ContainerListener listener)
```

Comment:
this should be `addContainerListener` otherwise it sounds like you are going to replace it.